carbonmark/components/Transaction/index.tsx

```diff
@@ -21,17 +20,25 @@ interface Props {
   approvalText?: React.ReactNode;
   submitText?: React.ReactNode;
   onGoBack?: () => void;
+  spenderAddress: string;
+  tokenIcon?: StaticImageData;
+  tokenName?: string;
+  onViewChange?: (newView: "approve" | "submit") => void;
+  showPrice?: boolean;
 }
 
 export const Transaction: FC<Props> = (props) => {
   const [view, setView] = useState<"approve" | "submit">(
     props.hasApproval ? "submit" : "approve"
   );
-
   const statusType = props.status?.statusType;
   const isPending =
     statusType === "userConfirmation" || statusType === "networkConfirmation";
 
+  useEffect(() => {
+    props.onViewChange?.(view);
```

Comment:
I am wondering why we need an external `onViewChange` function?

The component has the prop `hasApproval` 
=> if this is `true` => show "Submit" view, else show "Approve"

If a Transaction has an approval should be passed down by the corresponding view (Edit Listing, Create Listing, Purchase a Listing, and here => Retire Modal)